plugins/woocommerce/src/Internal/Admin/Orders/PostsRedirectionController.php

```diff
@@ -130,7 +130,16 @@ private function maybe_redirect_to_edit_order_page(): void {
 		$redirect_from_types   = wc_get_order_types( 'admin-menu' );
 		$redirect_from_types[] = 'shop_order_placehold';
 
-		if ( ! $post_id || ! in_array( get_post_type( $post_id ), $redirect_from_types, true ) || ! isset( $_GET['action'] ) ) {
+		$order_type = get_post_type( $post_id );
+		if ( ! $order_type ) {
+			// Attempt to determine whether this is a valid order on the HPOS side before bailing out.
+			$datastore = new \WC_Data_Store( 'order' );
```

Comment:
should this be `\WC_Data_Store::load( 'order' );`. It's the same things, but seems like we use this more often than the `new` keyword.

Reply:
[optional] Also maybe we can use `OrderUtil::is_order` or `OrderUtil::get_order_type` here? That way we will be using methods that are covered by unit tests, wdyt?

Reply:
That's a good idea. Thanks!